Tests/BSONTests/BSONObjectIDTests.swift

```diff
@@ -0,0 +1,81 @@
+@testable import BSON
+import Foundation
+import Nimble
+
+extension BSONObjectID {
+    // timestamp
+    internal var timestamp: Int {
```

Comment:
instead of all this bit shifting, I think it would be clearer to just use a temporrary `ByteBuffer` or `withUnsafeBytes` + `load` approach that we used somewhere else.